Assets/Scripts/PlayerManager.cs

```diff
@@ -99,6 +108,71 @@ private void Update()
         }
     }
 
+    private void FixedUpdate()
+    {
+        Vector3 displacement = Vector3.zero;
+
+        // Handle gravity
+        if (!m_CharacterController.isGrounded)
+        {
+            if (m_WasGrounded)
+            {
+                m_DownVelocity = Vector3.zero;
+                m_WasGrounded = false;
+            }
+            m_DownVelocity += m_Gravity * Time.fixedDeltaTime;
+            displacement = m_DownVelocity * Time.fixedDeltaTime;
+        } 
+        else
+        {
+            m_WasGrounded = true;
+        }
+
+        // Handle player input
+        displacement += m_DesiredDisplacement.normalized * m_WalkSpeed * Time.fixedDeltaTime;
+
+        // Handle platform movement
+        if (m_ActivePlatform != null)
+        {
+            // Determine where we should be standing
+            Vector3 newGlobalPoint = m_ActivePlatform.TransformPoint(m_LocalPlatformPoint);
+            // Determine how much to move the character to get there
+            Vector3 moveDistance = newGlobalPoint - m_GlobalPlatformPoint;
+            displacement += moveDistance;
+
+            // Determine where we should be standing
+            Quaternion newGlobalRotation = m_ActivePlatform.rotation * m_LocalPlatformRotation;
+            // Determine how much we should rotate to get there
+            Quaternion angleToRotate = newGlobalRotation * Quaternion.Inverse(m_GlobalPlatformRotation);
+            // Straighten the transform to point up
+            angleToRotate = Quaternion.FromToRotation(angleToRotate * transform.up, transform.up) * angleToRotate;
+            transform.rotation = angleToRotate * transform.rotation;
+        }
+
+        // Move the character
+        m_CharacterController.Move(displacement);
+
+        // Cleanup
+        m_DesiredDisplacement = Vector3.zero;
+        if (m_ActivePlatform != null)
+        {
+            m_GlobalPlatformPoint = transform.position;
+            m_LocalPlatformPoint = m_ActivePlatform.InverseTransformPoint(transform.position);
+            m_GlobalPlatformRotation = transform.rotation;
+            m_LocalPlatformRotation = Quaternion.Inverse(m_ActivePlatform.transform.rotation) * transform.rotation;
+        }
+    }
+
+    private void OnControllerColliderHit (ControllerColliderHit hit)
+    {
+        // ensure that the collision was down and beneath us
+        // ie we are we fell on and above the platform 
+        if (hit.moveDirection.y < -0.9 && hit.normal.y > 0.5)
```

Comment:
Those are normalized vectors representing the direction the character was moving during the collision and the normal of the surface the character hit. So the first number means 90% of the movement was in the negative y direction (down) and the second number means the surface normal is roughly in the up direction (we landed on top of  a slope of <=45 degrees).